src/main/scala/cromwell/engine/backend/jes/Run.scala

```diff
@@ -111,19 +83,52 @@ case class Run(jesId: String, pipeline: Pipeline, tag: String) {
     }
   }
 
+  private final def waitForStatus(breakout: RunStatus => Boolean): RunStatus = {
+
+    def checkStatus(previousStatus: Option[RunStatus]): RunStatus = {
+      val currentStatus = status()
+
+      if (!(previousStatus contains currentStatus)) {
+        // If this is the first time checking the status, we log the transition as '-' to 'currentStatus'. Otherwise
+        // just use the state names.
+        val prevStateName = previousStatus map { _.toString } getOrElse "-"
+        Log.info(s"$tag: Status change from $prevStateName to $currentStatus")
+
+        // Update the database state:
+        val newBackendInfo = JesCallBackendInfo(Option(JesId(jesId)), Option(JesStatus(currentStatus.toString)))
+        globalDataAccess.updateExecutionBackendInfo(workflowId, CallKey(call, pipeline.key.index), newBackendInfo)
```

Comment:
not related to your changes, but there should probably be a `recover` here to at least log if this `Future` fails